pkg/models/task_info.go

```diff
@@ -27,3 +31,44 @@ type TaskInfo struct {
 func (*TaskInfo) TableName() string {
 	return "tasks"
 }
+
+func (ti *TaskInfo) GetPulpData(pulpClient pulp_client.PulpClient) (api.PulpResponse, error) {
+	if ti.Typename == payloads.Snapshot {
+		var payload payloads.SnapshotPayload
+		response := api.PulpResponse{}
+
+		if err := json.Unmarshal(ti.Payload, &payload); err != nil {
+			return api.PulpResponse{}, errors.New("invalid snapshot payload")
+		}
+
+		if payload.SyncTaskHref != nil {
+			sync, syncErr := pulpClient.GetTask(*payload.SyncTaskHref)
+			if syncErr != nil {
+				return api.PulpResponse{}, syncErr
+			}
+			response.Sync = &api.PulpTaskResponse{}
+			api.ZestTaskResponseToApi(&sync, response.Sync)
+		}
+
+		if payload.DistributionTaskHref != nil {
+			distribution, distributionErr := pulpClient.GetTask(*payload.DistributionTaskHref)
+			if distributionErr != nil {
+				return api.PulpResponse{}, distributionErr
+			}
+			response.Distribution = &api.PulpTaskResponse{}
+			api.ZestTaskResponseToApi(&distribution, response.Distribution)
+		}
+
+		if payload.PublicationTaskHref != nil {
+			publication, publicationErr := pulpClient.GetTask(*payload.PublicationTaskHref)
+			if publicationErr != nil {
+				return api.PulpResponse{}, publicationErr
+			}
+			response.Publication = &api.PulpTaskResponse{}
+			api.ZestTaskResponseToApi(&publication, response.Publication)
+		}
+
+		return response, nil
+	}
+	return api.PulpResponse{}, errors.New("incorrect task type")
+}
```

Comment:
I don't know that the model package is the greatest place to put this method from a dependency perspective, as it makes the model package dependent on the pulp_client package.   I think the dao layer would be better (although even that isn't the best).